src/devices/Ft4222/Ft4222Spi.cs

```diff
@@ -0,0 +1,215 @@
+    {
+        private readonly SpiConnectionSettings _settings;
+        private IntPtr _ftHandle = new IntPtr();
+
+        /// <summary>
+        /// The connection settings of a device on a SPI bus. The connection settings are immutable after the device is created
+        /// so the object returned will be a clone of the settings object.
+        /// </summary>
+        public override SpiConnectionSettings ConnectionSettings => _settings;
+
+        /// <summary>
+        /// Store the FTDI Device Information
+        /// </summary>
+        public DeviceInformation DeviceInformation { get; internal set; }
+
+        /// <summary>
+        /// Create an SPI FT4222 class
+        /// </summary>
+        /// <param name="settings">SPI Connection Settings</param>
+        public Ft4222Spi(SpiConnectionSettings settings)
+        {
+            _settings = settings;
+            // Check device
+            var devInfos = FtCommon.GetDevices();
+            // Select the one from bus Id
+            DeviceInformation = devInfos[_settings.BusId];
+
+            // Open device
+            var ftStatus = FtFunction.FT_OpenEx(DeviceInformation.LocId, FtOpenType.OpenByLocation, ref _ftHandle);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to open device {DeviceInformation.Description}with error: {ftStatus}");
+
+            // Set the clock but we need some math
+            var (ft4222Clock, tfSpiDiv) = CalculateBestClockRate();
+
+            ftStatus = FtFunction.FT4222_SetClock(_ftHandle, ft4222Clock);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed set clock rate {ft4222Clock} on device: {DeviceInformation.Description}with error: {ftStatus}");
+
+
+            SpiClockPolarity pol = SpiClockPolarity.ClockIdleLow;
+            if ((_settings.Mode == SpiMode.Mode2) || (_settings.Mode == SpiMode.Mode3))
+                pol = SpiClockPolarity.ClockIdelHigh;
+
+            SpiClockPhase pha = SpiClockPhase.ClockLeading;
+            if ((_settings.Mode == SpiMode.Mode1) || (_settings.Mode == SpiMode.Mode3))
+                pha = SpiClockPhase.ClockTailing;
+
+            // Configure the SPI             
+            ftStatus = FtFunction.FT4222_SPIMaster_Init(_ftHandle, SpiOperatingMode.Single, tfSpiDiv, pol, pha, (byte)_settings.ChipSelectLine);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed setup SPI on device: {DeviceInformation.Description} with error: {ftStatus}");
+        }
+
+        private (FtClockRate clk, SpiClock spiClk) CalculateBestClockRate()
+        {
+            // Maximum is the System Clock / 1 = 80 MHz
+            // Minimum is the System Clock / 512 = 24 / 256 = 93.75 KHz
+            // Always take the below frequency to avoid over clocking
+            if (_settings.ClockFrequency < 187500)
+                return (FtClockRate.Clock24MHz, SpiClock.DivideBy256);
```

Comment:
should this be 48MHz?

Reply:
No, this one is correct. But, thanks for pointing out, the next one has to be 48.

Reply:
`24000000 / 256` shows 93750 for me which seems to be exactly half of the above number

Reply:
actually that's correct since it's `<`